src/Model.integration.test.ts

```diff
@@ -0,0 +1,244 @@
+import { Realtime } from 'ably/promises';
+import pino from 'pino';
+import { Subject } from 'rxjs';
+import { it, describe, expect, beforeEach, afterEach, vi } from 'vitest';
+
+import Model from './Model.js';
+import { defaultSyncOptions, defaultEventBufferOptions, defaultOptimisticEventOptions } from './Options.js';
+import type { ModelOptions } from './types/model.js';
+import { createAblyApp } from './utilities/test/createAblyApp.js';
+import { getEventPromises } from './utilities/test/promises.js';
+
+interface TestContext extends ModelOptions {
+  model: Model<any>;
+  channelName: string;
+  eventData: {
+    name: string;
+    data: any;
+    mutationId: string;
+  };
+}
+
+describe('Stream integration', () => {
+  beforeEach<TestContext>(async (context) => {
+    const channelName = 'test-channel';
+    const data = await createAblyApp({
+      keys: [{}],
+      namespaces: [{ id: channelName, persisted: true }],
+      channels: [
+        {
+          name: channelName,
+          presence: [
+            { clientId: 'John', data: 'john@test.com' },
+            { clientId: 'Dave', data: 'dave@test.com' },
+          ],
+        },
+      ],
+    });
+    const ably = new Realtime({
+      key: data.keys[0].keyStr,
+      environment: 'sandbox',
+    });
+    const logger = pino({ level: 'debug' });
+    const model = new Model(
+      channelName,
+      {
+        sync: async (data) => data,
+        merge: (state, event) => (state ? { ...state, ...event.data } : event.data),
+      },
+      {
+        ably,
+        channelName,
+        logger,
+        syncOptions: defaultSyncOptions,
+        optimisticEventOptions: defaultOptimisticEventOptions,
+        eventBufferOptions: defaultEventBufferOptions,
+      },
+    );
+
+    context.model = model;
+    context.channelName = channelName;
+    context.ably = ably;
+
+    context.eventData = {
+      name: 'update',
+      data: {
+        noOfMessages: 34,
```

Comment:
This might be a confusing property, as 34 messages are never sent; would be better to just call the key `foo`